Use also yaml files as configuration in `src/vars`

leophys opened the pull request:
This is a small PR to allow to use yaml files as configuration files for mailbundle.

I tested it with a fairly complex multi-account configuration file. This PR also provides the default example configuration, translated from `.json` to `.yaml`.

blallo committed:
Allow yaml to be used as configuration files' format

blallo committed:
Update afew version

blallo committed:
Update requirements.txt

blallo committed:
Merge branch 'master' of https://github.com/boyska/mailbundle

blallo committed:
Adding yaml conf example

boyska commented:
just 2 issues and 1 (positive) critique!

### Optional dependency

this should be trivial to fix. I'd like not to add a requirement on `yaml` library, unless there is indeed a `*.yaml` file in my `src/vars/` directory.

### Sorting

what about sorting? what is the expected behavior if I have both `15-accounts.json` and `15-accounts.yml` ?
I expect this to be either:
* an error (preferred! sth wrong is taking place when you have both)
* a warning + clear documentation

### Why?

Do you find yaml easy? cause I don't :)

If I had to add one more file format for 1) easier human annotation 2) clarity 3) technical features, TOML would be that format. See https://pypi.org/project/toml/ 
I think adding TOML support instead of yaml is equally easy.

For an example, I think that this would be very useful: [TOML array of tables](https://github.com/toml-lang/toml#array-of-tables)

```toml
[[accounts]]
name = "Foo"
type = "gmail"
email = "foo@gmail.com"

[[accounts]]
name = "j0hn"
imap_host = "imap.autistici.org"
```

etc.

blallo committed:
Improved support for yaml and added for toml

blallo committed:
Merge branch 'master' of https://github.com/boyska/mailbundle

blallo committed:
Fixing dependencies

leophys commented:
Hi. As you suggested, I added support for `.toml` file format.

I handled the two issues you pointed out in the following way.
## Optional depenencies
The function `read_conf` now first searches all the config files in `src/vars`. If there is any `.yaml` or `.yml` that follows the naming convention (two digits numbers at the beginning) it imports the `yaml` module.
The very same thing is done for the `.toml` files, in case there is any.

## Sorting
Now the function logs an ERROR, pointing out the first conflict it finds (two files with the same numeration). and raises a `ValueError`. To keep simple this check, I moved all the examples in `src/vars/examples/`. And also provided a correspondent `.yaml` and `.toml` for any `.json` example you provided.

## Why
I like `YAML`. I find it very easily readable :) I suppose it's just a matter of taste.

blallo committed:
Update HOWTO and examples

blallo committed:
Update conf file logic

leophys commented:
Again, another update :)

### The sorting logic

I changed the whole logic of the `read_conf`. Not it does not give an error when there are two or more files that begin with the same number, but only when two (or more) files differ only by the extension.
Namely, the following is allowed:
```
10-a_conf.yaml
10-b_conf.yaml
10-c_conf.json
10-d_conf.toml
```
This is *not* allowed:
```
10-conf.json
10-conf.yaml
```
This is also allowed but, as in the original code, outputs a warning:
```
10-a_conf.yaml
another_conf.json
```

### The optional dependencies

As you suggested privately, I moved the import of `yaml` and `toml` modules as near as their usage as possible. The puzzling thing here is that I placed them as mandatory dependencies in `src/requirements.txt`, so I don't really see the point in not importing these modules from the very beginning.
